src/arrayMethodJoin.js

```diff
@@ -5,7 +5,62 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
```

Comment:
you can set separator by default
```suggestion
  [].__proto__.join2 = function(separator = ',') {
```

Reply:
> 



> GJ! But your solution is too long. Please rewrite it using `for` loop
Can I get a tip on exactly how to use `for` loop , to shorten the code???

Reply:
давай спростимо існуючий цикл
задамо дві умови: одна для `this[i] `: добавляємо до строки якщо це не `null `або `undefined`
друга умова добавляємо `separator` завжди, крім випадку коли наш` i `- це останній елемент` this`

два` if` в циклі і все

Reply:
> 

Ну я дуже довго сидів над цим і переписував по 10 раз, все що в мене вийшло це ось так
Якщо забрати ось цей 
        if (separator === null && i !== this.length - 1) {
          str += this[i] + 'null';
          continue;
        }
тоді всі тести проходять окрім одного "Should work with null"